src/helpers/all.php

```diff
@@ -169,12 +169,16 @@ function create_uri_from_global()
 
             if (isset($matches[2])) {
                 $port = (int)substr($matches[2], 1);
+            } else {
+                $port = $isSecure ? 443 : 80;
             }
         } else {
             $pos = strpos($host, ':');
             if ($pos !== false) {
                 $port = (int)substr($host, $pos + 1);
                 $host = strstr($host, ':', true);
+            } else {
+                $port = $isSecure ? 443 : 80;
             }
         }
 
```

Comment:
I have no way to test this out at the moment, but I believe in this case it will be best to check the condition outside.

Example: 
```
if ($isSecure && $port === 80) {
    $port = 443;
}
```